docs/06-contributors/999-rfcs/000-stories/story-13.md

```diff
@@ -0,0 +1,169 @@
+bring cloud;
+// need to run: npm install uuid --save
+bring uuid;
+
+/**
+ * Represents a cloud task list.
+ */
+resource TaskList {
+  /** stores the tasks */
+  _bucket: cloud.Bucket;
+  
+  init() {
+    this._bucket = new cloud.Bucket();
+  }
+
+  /** 
+   * Adds a task to the task list.
+   * @returns The ID of the new task.
+   */
+  inflight add_task(title: str): str {
+    let id = uuid.v4();
+    let j = Json { 
+      title: title, 
+    };
+    print("adding task ${id} with data: ${j}"); //j should be printed out nicely 
+    this._bucket.put(id, j.to_str());
+    return id;
+  }
+
+  /** 
+   * Gets a task from the task list.
+   * @param id - the id of the task to return
+   * @returns the title of the task (optimistic)
+   */
+  inflight get_task(id: str): Json {
+    let blob = this._bucket.get(id);
+    return Json.parse(blob)
+  }
+
+  /** 
+   * Sets effort estimation on a test
+   * @param id - the id of the task to return
+   * @param effort_estimation - the time (duration) estimated for this task
+   * @returns The ID of the existing task.
+   */
+  inflight add_estimation(id: str, effort_estimation: duration) {
+    let j = this.get_task(id).to_mut();
+    j.set("effort_estimation", effort_estimation);
+    this._bucket.put(id, j.to_str());
+    return id;
+  }
+
+  /** 
+   * Removes a task from the list
+   * @param id - the id of the task to be removed
+   */
+  inflight remove_tasks(id: str): str {
+    print("removing task ${id}");
+    this._bucket.delete(id);
+    return id;
+  }
+
+   /** 
+    * Gets the tasks ids 
+    * @returns set of task id
+    */
+  inflight list_task_ids(): Set<str> {
+    let result = MutSet<str> {};
+    for id in this._bucket.list() {
+      result.add(id);
+    }
+    return result.to_immut();
+  }
+
+   /** 
+    * Find tasks with title that contains a term
+    * @param term - the term to search
+    * @returns set of task id that matches the term
+    */
+  inflight find_tasks_with(term: str): Array<str> {
+    print("find_tasks_with: ${term}");
+    let task_ids = this.list_task_ids();
+    print("found ${task_ids.size} tasks");
+    let output = MutArray<str>[];// #1172
+    for id in task_ids {
+      let j = this.get_task(id); 
+      let title = Str.from_json(j.get("title"));
+      if title.contains(term) { 
+        print("found task ${id} with title \"${title}\" with term \"${term}\"");
+        output.push(id);
+      }
+    }
+    
+    print("found ${output.len} tasks which match term '${term}'");
+    return output.to_immut();
+  }
+}
+
+// --------------------------------------------
+// testing
+// --------------------------------------------
+
+resource Test {
+  tasks: TaskList; 
+  _test: inflight (Test): void;
+
+  inflight before() {
+    for id in this.tasks.list_task_ids() {
+      this.tasks.remove_tasks(id);
+    }
+  }
+
+  init(name: str, test: inflight (Test): void) {
+    this.tasks = new TaskList();
+    this._test = test;
+    new cloud.Function({
```

Comment:
This really feels like a (very nice) hack... let's think how we can make this a little nicer...

Reply:
We are still hacking this for now. It turns out that this hack will not work for this sprint... 

I'll have to change this for now (probably tomorrow when we know what is actually in)

Reply:
Why do we need this actually? Why not just a bunch of tests?

Reply:
moving them to simple `cloud.Function` tests

Reply:
done